### PR TITLE
Enable generic external_custom trigger

### PR DESCRIPTION
**Pull request by SAY-droid427**
I have created a draft for enabling generic external_custom target.

### CODE REVIEW

**Comment by SAY-droid427** [comment]
I tried to follow the flow of running external tests by
```
export BUILD_LIST=external/external_custom
make compile
```
but I got the following output
```
    [exec]  ---> bdac1d1b0d29
     [exec] Step 16/26 : COPY derby/dockerfile/derby-test.sh /derby-test.sh
     [exec] COPY failed: file not found in build context or excluded by .dockerignore: stat derby/dockerfile/derby-test.sh: file does not exist

BUILD FAILED
/home/sayani/Desktop/openjdk-tests/TKG/scripts/build_test.xml:76: The following error occurred while executing this line:
/home/sayani/Desktop/openjdk-tests/external/external_custom/build.xml:25: The following error occurred while executing this line:
/home/sayani/Desktop/openjdk-tests/external/build.xml:87: exec returned: 1

Total time: 5 minutes 4 seconds
1
compile.mk:32: recipe for target 'compile' failed
make[1]: *** [compile] Error 1
make[1]: Leaving directory '/home/sayani/Desktop/openjdk-tests/TKG'
makefile:59: recipe for target 'compile' failed
make: *** [compile] Error 2
sayani@sayani-VivoBook-ASUSLapt
```
However, I can run other tests successfully.
```
dist:
     [copy] Copying 2 files to /home/sayani/Desktop/jvmtest/external/jacoco

BUILD SUCCESSFUL
Total time: 5 minutes 43 seconds
0
make[1]: Leaving directory '/home/sayani/Desktop/openjdk-tests/TKG'

```

File: external/external_custom/build.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<?xml version="1.0"?>
+<project name="External_custom Test" default="build" basedir=".">
+	<taskdef resource="net/sf/antcontrib/antlib.xml" />
+	<description>
+		Build external_custom Docker image
+	</description>
+	<import file="${TEST_ROOT}/external/build.xml"/>
+
+	<!-- set properties for this build -->
+	<property name="TEST" value="derby" />
```
Comment (smlambert) [inline external/external_custom/build.xml:10]: derby needs to be changed to external_custom

Comment (SAY-droid427) [reply external/external_custom/build.xml:10]: Yeah I will change it and try again

Comment (smlambert) [reply external/external_custom/build.xml:10]: will need something like: 
```
<property environment="env" />
<property name="EXT_CUSTOM_TARGET" value=${env.CUSTOM_EXTERNAL_TARGET}
```

Comment (SAY-droid427) [reply external/external_custom/build.xml:10]: I have used
```
<property environment="env" />
<property name="EXTERNAL_CUSTOM_TARGET" value="${env.EXTERNAL_CUSTOM_TARGET}" />
``` 
but the build fails. 

For reading the test.properties file we can follow one of the following methods in the common_function
```while IFS= read -r line
do
    echo $line
done < external_custom/test.properties
```
or
```
cat external_custom/test.properties | while read line
do 
    echo $line
done
```

Comment (SAY-droid427) [reply external/external_custom/build.xml:10]: For reading the test.properties file we can follow one of the following methods in the common_function
```
while IFS= read -r line
do
    echo $line
done < external_custom/test.properties
```
or
```
cat external_custom/test.properties | while read line
do 
    echo $line
done
```

**Comment by SAY-droid427** [comment]
@smlambert I have added the code for reading individual properties from test.properties file in a function getProperty() and I have called the function from external_custom. But I am not sure whether that is what  had to be done. Another question is that the external_custom directory has no test.properties file, so do we have to create that explicitly?

**Comment by SAY-droid427** [comment]
I am closing this because I accidentally rebased the commit and included other commits.

**Pull request closed without merging**